### PR TITLE
Add namespace support

### PR DESCRIPTION
**Pull request by idontwanttosignin**
Added optional namespace field and conditionally pass it to the driver
if not empty (passing an empty namespaces errors).

### CODE REVIEW

**Comment by maheshp** [comment]
@idontwanttosignin I am taking a look at the PR, will get back to you soon if I have any questions.

File: src/main/java/com/thoughtworks/gocd/secretmanager/vault/builders/VaultConfigBuilder.java [hunk 1]
```diff
@@ -26,12 +26,14 @@
...
     public VaultConfig configFrom(SecretConfig secretConfig) throws VaultException {
-        return new VaultConfig()
+        VaultConfig request = new VaultConfig()
                 .address(secretConfig.getVaultUrl())
                 .openTimeout(secretConfig.getConnectionTimeout())
                 .readTimeout(secretConfig.getReadTimeout())
-                .sslConfig(sslConfig(secretConfig).build())
-                .build();
+                .sslConfig(sslConfig(secretConfig).build());
+        if (isNotBlank(secretConfig.getNameSpace()))
```
Comment (maheshp) [inline src/main/java/com/thoughtworks/gocd/secretmanager/vault/builders/VaultConfigBuilder.java]: Can you please add a test for these changes.

Comment (idontwanttosignin) [reply src/main/java/com/thoughtworks/gocd/secretmanager/vault/builders/VaultConfigBuilder.java]: Is there something specific you recommend testing? Happy to throw in a bit more to the tests, but not really sure what we're looking for.

Comment (idontwanttosignin) [reply src/main/java/com/thoughtworks/gocd/secretmanager/vault/builders/VaultConfigBuilder.java]: Is there something specific you recommend testing? Happy to throw in a bit more to the tests, but not really sure what we're looking for.

Comment (maheshp) [reply src/main/java/com/thoughtworks/gocd/secretmanager/vault/builders/VaultConfigBuilder.java]: I would verify that the Builder would return a VaultConfig with namespace if provided, something similar to - https://github.com/gocd/gocd-vault-secret-plugin/blob/master/src/test/java/com/thoughtworks/gocd/secretmanager/vault/builders/AppRoleVaultConfigBuilderTest.java#L34

**Pull request merged**